### PR TITLE
Add zh-HK translation.json

### LINKED ISSUE #882: Add zh-Hant translation.json
This is a translation contributed by @jimcky 

```json
{
  "activated-label": "Activated",
  "add-button-label": "Add",
  "app.name": "Authgear",
  "apple-app-store-label": "Apple App Store",
  "back-button-title": "返回",
  "change-button-label": "Change",
  "change-password-button-label": "Change Password",
  "change-password-page-description": "Please enter your new password below.",
  "change-password-page-title": "Change Password",
  "change-secondary-password-page-description": "Please enter your new password below.",
  "change-secondary-password-page-title": "Change Additional Password",
  "confirm-password-placeholder": "再次輸入密碼",
  "connect-button-label": "Connect",
  "create-device-token-label": "Do not ask again on this device",
  "create-password-page-title": "建立密碼",
  "details-button-label": "Details",
  "disconnect-button-label": "Disconnect",
  "download-button-label": "Download",
  "email-placeholder": "電郵地址",
  "email.authenticate-primary-oob.subject": "Logging in to {AppName}",
  "email.authenticate-secondary-oob.subject": "Logging in to {AppName}",
  "email.default.reply-to": "",
  "email.default.sender": "no-reply@authgear-staging.com",
  "email.default.subject": "",
  "email.forgot-password.subject": "Reset password instruction",
  "email.setup-primary-oob.subject": "Signing up at {AppName}",
  "email.setup-secondary-oob.subject": "Signing up at {AppName}",
  "email.verification.subject": "Email Verification Instruction",
  "email.welcome-message.subject": "Welcome!",
  "enter-login-id-page-description--change": "Your current {key} is <b>{displayID}</b>. Please enter your new {key}.",
  "enter-login-id-page-title--add": "Enter your {key}",
  "enter-login-id-page-title--change": "Change your {key}",
  "enter-oob-otp-instead": "Send code to {target} instead",
  "enter-password-instead": "Enter password instead",
  "enter-password-page-title": "Enter password",
  "enter-recovery-code-description": "Enter any unused 10-letter recovery code",
  "enter-recovery-code-instead": "Enter recovery code instead",
  "enter-recovery-code-placeholder": "10-letter recovery code",
  "enter-recovery-code-title": "2-step verification",
  "enter-totp-code-placeholder": "6-digit code from the authenticator app / device",
  "enter-totp-description": "Enter the 6-digit code you see in the authenticator app / device",
  "enter-totp-instead": "Enter 6-digit authenticator code instead",
  "enter-totp-title": "2-step verification",
  "error-blocked-user": "Your account is blocked from login.",
  "error-calling-code-required": "Calling code is required",
  "error-code-required": "Code is required",
  "error-confirm-password-required": "請輸入確認密碼",
  "error-disabled-user": "Administrator has disabled your account.",
  "error-disabled-user-no-reason": "Please contact administrator for details.",
  "error-disabled-user-reason": "Reason: {reason}",
  "error-duplicated-authenticator": "This authenticator already has been added.",
  "error-duplicated-identity": "This identity has been claimed by another user.",
  "error-email-required": "Email is required",
  "error-invalid-credentials": "Invalid credentials",
  "error-invalid-email": "Invalid email address",
  "error-invalid-username": "Invalid username",
  "error-login-id-required": "{variant, select, email {電郵地址} username {使用者名稱} other {電郵地址或使用者名稱}} is required",
  "error-max-length": "Expect string with length less than or equal to {expected, plural, one{# character} other{# characters}}",
  "error-network": "There is a problem with the network.",
  "error-new-password-required": "新密碼為必填項目",
  "error-new-password-typo": "確認密碼不正確",
  "error-old-password-required": "Old password is required",
  "error-password-required": "Password is required",
  "error-password-reset-failed": "This reset password link is invalid, used or expired. Please request a new one.",
  "error-phone-number-format": "Invalid phone number",
  "error-phone-number-required": "Phone is required",
  "error-rate-limited": "因偵測到過於頻繁的使用，此功能已被暫時停用。請稍侯片刻後重試。",
  "error-remove-last-identity": "Cannot disconnect. You need to keep at least 1 identity.",
  "error-remove-last-primary-authenticator": "Cannot remove. You need to keep at least 1 primary authenticator for an identity.",
  "error-remove-last-secondary-authenticator": "Cannot remove. Multi-factor authentication is required.",
  "error-server": "Server encountered an internal error.",
  "error-web-ui-invalid-session-action": "Return",
  "error-web-ui-invalid-session-retry": "<p>This page has expired/invalid, maybe you have idled too long or switched your browser?</p><p>Please return and retry your request.</p>",
  "error-web-ui-invalid-session-return": "<p>This page is no longer active. Please return to the current page.</p>",
  "forgot-password-button-label": "無法登入你的帳戶?",
  "forgot-password-button-label--enter-password-page": "Forgot Password?",
  "forgot-password-email-description": "Enter your email to request instruction to reset your password",
  "forgot-password-page-title": "Forgot Password",
  "forgot-password-phone-description": "Enter your phone to request instruction to reset your password",
  "forgot-password-send-to-email-description": "Request to send reset password instruction to the email below",
  "forgot-password-send-to-phone-description": "Request to send reset password instruction to the phone below",
  "forgot-password-success-description": "If you have an account, please follow the instruction sent to {target} to reset your password",
  "forgot-password-success-page-title": "Request received",
  "get-new-code-button-label": "Get new code",
  "google-play-store-label": "Google Play Store",
  "hide-password-label": "Hide password",
  "inactive-label": "Inactive",
  "login-button-label--forgot-password-success-page": "Sign in",
  "login-id-placeholder": "{variant, select, email {電郵地址} username {使用者名稱} other {電郵地址或使用者名稱}}",
  "login-label": "登入",
  "logout-button-hint": "Click the button to sign out",
  "logout-button-label": "Sign out",
  "more-button-label": "More",
  "new-password-placeholder": "新密碼",
  "next-button-label": "繼續",
  "oauth-branding-apple": "Sign in with Apple",
  "oauth-branding-azureadv2": "Sign in with Microsoft",
  "oauth-branding-facebook": "使用 Facebook 帳戶登入",
  "oauth-branding-google": "使用 Google 帳戶登入",
  "oauth-branding-linkedin": "Sign in with LinkedIn",
  "oauth-branding-wechat": "Login with WeChat",
  "oob-otp-description": "我們已發送一段 {length}位的認證碼到 {target}. 請輸入該認證碼後繼續下一步",
  "oob-otp-page-title--email": "Email One-Time-Password",
  "oob-otp-page-title--sms": "SMS One-Time-Password",
  "oob-otp-placeholder": "認證碼",
  "oob-otp-resend-button-hint": "未收到認證碼? ",
  "oob-otp-resend-button-label": "重發",
  "oob-otp-resend-button-label--unit": "重發 (%ds)",
  "oob-otp-send-description": "Click next to get the code at {target}.",
  "password-placeholder": "Current password",
  "password-policy-banned-words": "NO banned words",
  "password-policy-digit": "At least 1 digit",
  "password-policy-lowercase": "At least 1 lowercase character",
  "password-policy-minimum-length": "最少 {length, plural, one{# character} other{# 位}}",
  "password-policy-password-strength-label": "密碼強度:",
  "password-policy-password-strength-meter-0": "無保護性",
  "password-policy-password-strength-meter-1": "非常容易破解",
  "password-policy-password-strength-meter-2": "容易破解",
  "password-policy-password-strength-meter-3": "一般",
  "password-policy-password-strength-meter-4": "不易破解",
  "password-policy-password-strength-meter-5": "非常不易破解",
  "password-policy-required-password-strength-label": "Your new password must be",
  "password-policy-reuse": "No reuse of {size, plural, one{# previous password} other{# previous passwords}} / previous password within {day, plural, one{# day} other{# days}}",
  "password-policy-symbol": "At least 1 symbol",
  "password-policy-title": "你的密碼必須符合以下要件:",
  "password-policy-uppercase": "At least 1 uppercase character",
  "phone-number-placeholder": "phone",
  "privacy-policy-link": "https://www.authgear.com/data-privacy",
  "public-signup-disabled-notice": "Please contact the system administrator for creating new accounts.",
  "recovery-code-consumption-description": "You can only use each code once",
  "recovery-code-list-disabled": "Listing recovery code is disabled by administrator",
  "recovery-code-storage-description": "Keep these recovery codes somewhere safe but accessible",
  "recovery-code-title": "Recovery Code",
  "remove-button-label": "Remove",
  "resend-verification-code-button-label": "Resend verification code",
  "reset-password-description": "Please enter your new password below.",
  "reset-password-page-title": "Reset Password",
  "reset-password-success-description": "You have successfully reset your password. You can now sign in with it.",
  "reset-password-success-page-title": "Password Reset",
  "return-page-description": "Please return to where you were to continue. If you were interacting with an mobile app, please switch back to the app. If you were interacting with a website, please switch back to that tab.",
  "return-page-title": "Return to where you were",
  "revoke-all-device-tokens-button-label": "Forget all trusted devices",
  "settings-identity-description": "There are the methods to sign in your account",
  "settings-identity-login-id-email": "Email Address",
  "settings-identity-login-id-phone": "Phone Number",
  "settings-identity-login-id-raw": "Username",
  "settings-identity-login-id-username": "Username",
  "settings-identity-oauth-apple": "Apple",
  "settings-identity-oauth-azureadv2": "Azure AD",
  "settings-identity-oauth-facebook": "Facebook",
  "settings-identity-oauth-google": "Google",
  "settings-identity-oauth-linkedin": "LinkedIn",
  "settings-identity-oauth-wechat": "WeChat",
  "settings-identity-title": "Login Info",
  "settings-oob-otp-item-description": "Added at {time, datetime, long}",
  "settings-oob-otp-title": "Channels for receiving verification code via SMS / Email",
  "settings-page-mfa-section-description": "Manage your list of multi-factor authenticators.",
  "settings-page-mfa-section-title": "Multi-factor Verifications",
  "settings-page-primary-oob-otp-description": "Added at {time, datetime, long}",
  "settings-page-primary-oob-otp-title--email": "Passwordless via Email",
  "settings-page-primary-oob-otp-title--sms": "Passwordless via SMS",
  "settings-page-primary-password-description": "Last updated at {time, datetime, long}",
  "settings-page-primary-password-title": "Password",
  "settings-page-recovery-code-description": "These codes allow you to sign in even you lost access to your apps, devices, phones or email accounts.",
  "settings-page-recovery-code-title": "Recovery Code",
  "settings-page-secondary-oob-otp-title": "Verification code via SMS / Email",
  "settings-page-secondary-password-description": "Last updated at {time, datetime, long}",
  "settings-page-secondary-password-title": "Additional Password",
  "settings-page-secondary-totp-title": "Authenticator Apps / Devices",
  "settings-page-security-section-description": "These can be used to make sure that it''s really you signing in.",
  "settings-page-security-section-title": "Security",
  "settings-page-session-section-description": "Manage your signed in sessions on your autenticated devices.",
  "settings-page-session-section-title": "Signed in Sessions",
  "settings-sessions-description": "You’re currently signed in to your account on these devices",
  "settings-sessions-item-description": "Last activity: {time, datetime, long}",
  "settings-sessions-revoke-all-label": "Terminate all other sessions",
  "settings-sessions-revoke-label": "Terminate",
  "settings-sessions-title": "Signed in Sessions",
  "settings-totp-item-description": "Added at {time, datetime, long}",
  "settings-totp-title": "Authenticator Apps / Devices",
  "setup-button-label": "Setup",
  "setup-oob-otp-instead": "Setup SMS / email 2-step verification instead",
  "setup-oob-otp-title--email": "Setup email 2-step verification",
  "setup-oob-otp-title--sms": "Setup SMS 2-step verification",
  "setup-oob-otp-use-email-description": "Setup email 2-step verification instead",
  "setup-oob-otp-use-phone-description": "Setup SMS 2-step verification instead",
  "setup-password-button-label": "Setup Password",
  "setup-password-instead": "Setup password instead",
  "setup-totp-code-description": "Enter the 6-digit code that you see in the app / device",
  "setup-totp-code-placeholder": "code",
  "setup-totp-fallback-description": "Can''t scan the barcode?\nChoose <b>Manual entry</b>.\nEnter your email address and this key:\n<b>{secret}</b>\nMake sure that Time based is turned on",
  "setup-totp-get-google-authenticator-description": "If you don''t have any authenticator device / app, you can get the Google Authenticator App from <a target=\"_blank\" href=\"https://play.google.com/store/apps/details?id=com.google.android.apps.authenticator2\">Google Play Store</a> or <a target=\"_blank\" href=\"https://apps.apple.com/app/google-authenticator/id388497605\">Apple App Store</a>",
  "setup-totp-instead": "Setup authenticator app / device instead",
  "setup-totp-title": "Setup authenticator",
  "setup-totp-use-google-authenticator-description": "In the App select <b>Set up account</b>.\nChoose <b>Scan barcode</b>",
  "show-password-label": "Show password",
  "show-recovery-code-button-label": "Show Codes",
  "signup-label": "註冊",
  "sms.default.sender": "Authgear",
  "sso-login-id-separator-both-present": "或",
  "sso-login-id-separator-oauth-present": "Choose an option below to continue",
  "terms-of-service-link": "https://www.authgear.com/terms",
  "toc-pp-footer": "註冊即表示我已同意 <a target=\"_blank\" href=\"{termsOfService}\">條款和細則</a> 及 <a target=\"_blank\" href=\"{privacyPolicy}\">隱私政策</a>.",
  "use-email-login-id-description": "使用電郵登入",
  "use-login-id-key": "Use {key} instead",
  "use-phone-login-id-description": "使用電話號碼登入",
  "use-text-login-id-description": "Use {variant, select, email {電郵地址} username {使用者名稱} other {電郵地址 or 使用者名稱}} instead",
  "verification-status-label--unverified": "Not verified",
  "verification-status-label--verified": "Verified",
  "verify-user-description": "我們已發送一段 {length}位的認證碼到 {target}. 請輸入該認證碼後繼續下一步",
  "verify-user-page-title--email": "認證你的電郵地址",
  "verify-user-page-title--sms": "短訊認證",
  "verify-user-placeholder": "認證碼",
  "verify-user-resend-button-hint": "未收到認證碼?",
  "verify-user-resend-button-label": "重發",
  "verify-user-resend-button-label--unit": "重發 (%ds)",
  "verify-user-success-description": "你的帳號已完成驗證",
  "verify-user-success-page-title": "帳號驗證",
  "wechat-auth-title": "Login with WeChat",
  "wechat-auth-with-app-description": "Click the below button to continue WeChat Login",
  "wechat-auth-with-qr-code-description": "Please scan the QR code in your WeChat app to sign in.",
  "wechat-open-app": "Open WeChat app to login",
  "wechat-proceed": "Proceed WeChat login"
}
```

### PR DESCRIPTION
**Pull request by louischan-oursky**
fix #882

### CODE REVIEW

File: resources/authgear/templates/zh-HK/translation.json [hunk 1]
```diff
@@ -0,0 +1,229 @@
+{
+  "activated-label": "已啟用",
+  "add-button-label": "增加",
+  "app.name": "Kurious",
```
Comment (carmenlau) [inline resources/authgear/templates/zh-HK/translation.json:4]: incorrect app name

File: resources/authgear/templates/zh-HK/translation.json [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+  "create-device-token-label": "不要再詢問",
+  "create-password-page-title": "建立密碼",
+  "details-button-label": "查看",
+  "disconnect-button-label": "移除",
+  "download-button-label": "下載",
+  "email-placeholder": "電郵地址",
+  "email.authenticate-primary-oob.subject": "登入 {AppName}",
+  "email.authenticate-secondary-oob.subject": "登入 {AppName}",
+  "email.default.reply-to": "",
+  "email.default.sender": "no-reply@kurious.live",
```
Comment (carmenlau) [inline resources/authgear/templates/zh-HK/translation.json:26]: should be no-reply@authgear.com

File: resources/authgear/templates/zh-HK/translation.json [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+  "password-policy-password-strength-meter-3": "一般",
+  "password-policy-password-strength-meter-4": "不易破解",
+  "password-policy-password-strength-meter-5": "非常不易破解",
+  "password-policy-required-password-strength-label": "Your new password must be",
+  "password-policy-reuse": "No reuse of {size, plural, one{# previous password} other{# previous passwords}} / previous password within {day, plural, one{# day} other{# days}}",
+  "password-policy-symbol": "At least 1 symbol",
+  "password-policy-title": "你的密碼必須符合以下條件:",
+  "password-policy-uppercase": "At least 1 uppercase character",
+  "phone-number-placeholder": "phone",
+  "privacy-policy-link": "https://kurious.live/zh/privacy-policy",
```
Comment (carmenlau) [inline resources/authgear/templates/zh-HK/translation.json:133]: update the link

File: resources/authgear/templates/zh-HK/translation.json [hunk 4]
```diff
@@ -0,0 +1,229 @@
...
+  "setup-totp-instead": "Setup authenticator app / device instead",
+  "setup-totp-title": "Setup authenticator",
+  "setup-totp-use-google-authenticator-description": "In the App select <b>Set up account</b>.\nChoose <b>Scan barcode</b>",
+  "show-password-label": "顯示密碼",
+  "show-recovery-code-button-label": "顯示代碼",
+  "signup-label": "註冊",
+  "sms.default.sender": "Authgear",
+  "sso-login-id-separator-both-present": "或",
+  "sso-login-id-separator-oauth-present": "選擇以下一個選項以繼續",
+  "terms-of-service-link": "https://kurious.live/zh/terms-and-conditions",
```
Comment (carmenlau) [inline resources/authgear/templates/zh-HK/translation.json:207]: update the link

File: resources/authgear/templates/zh-HK/translation.json [hunk 5]
```diff
@@ -0,0 +1,229 @@
...
+  "toc-pp-footer": "註冊即表示我已同意 <a target=\"_blank\" href=\"{termsOfService}\">條款和細則</a> 及 <a target=\"_blank\" href=\"{privacyPolicy}\">隱私政策</a>.",
```
Comment (carmenlau) [inline resources/authgear/templates/zh-HK/translation.json:208]: Although we support theme primary color for anchor without class. We may want to sync the class in 2 different locale, maybe remove the class in EN version?

Comment (louischan-oursky) [reply resources/authgear/templates/zh-HK/translation.json]: I will add the class to the zh version.

**Review by carmenlau — COMMENTED** [review]
@louischan-oursky I just tested the portal and find this issue may not as simple as adding the translation.json.

- Developer cannot remove the Chinese language from the portal
- Developer cannot set Chinese as default language, the portal ui allows him to do it but it will fail because there is only translation.json but no other templates.  

Also, for other non-default languages, maybe we should remove some items (`email.default.sender`, `email.default.reply-to`, `sms.default.sender`... and others which have similar case). So in most of the case,  if developer wants to override the value, they only need to set it once?

**Comment by louischan-oursky** [comment]
> Developer cannot remove the Chinese language from the portal

This reveals that the developer cannot remove the English language from the portal. Right now we derive supported languages from available templates. So you are correct, once we add more builtin supported languages, builtin supported languages cannot be removed. Let's discuss this in the meeting.

> Developer cannot set Chinese as default language, the portal ui allows him to do it but it will fail because there is only translation.json but no other templates.

What do you mean by fail? Do you mean the end user will still see english email and SMS messages?

> Also, for other non-default languages, maybe we should remove some items (email.default.sender, email.default.reply-to, sms.default.sender... and others which have similar case). So in most of the case, if developer wants to override the value, they only need to set it once?

Make sense.

---

It seems that we better not do this for now...

**Comment by carmenlau** [comment]
> What do you mean by fail?

The UI allows me to click change default language to Chinese, but will get error **Some of the languages are invalid. Some required resources do not have localizations in default language**.

> It seems that we better not do this for now...

Agree. Need more discussion on handling supported languages

**Pull request closed without merging**